crypto/fipsmodule/self_check/self_check.c

```diff
@@ -927,6 +927,69 @@ static int boringssl_self_test_hkdf_sha256(void) {
                     "HKDF-SHA-256 KAT");
 }
 
+static int boringssl_self_test_evp_hkdf_sha256(void) {
```

Comment:
This self-test is not needed since the evp API calls the same low level function `HKDF()`. We usually want to do the minimum in self_test.

Reply:
I thought we'd decided we need to test all the APIs we claim as validated?

Reply:
We can have as many tests as we want/should in the actual tests (unit, ACVP, service indicator), but when it comes to the self-test, we want to test the minimum number of APIs that covers the actual crypto implementation we're testing. So, for example, ECDSA is not tested via EVP_DigestSign(), neither is RSA, but with special functions that go directly to the low-level code.